L1Trigger/TrackTrigger/src/L1TrackQuality.cc

```diff
@@ -118,12 +118,14 @@ std::vector<float> L1TrackQuality::featureTransform(TTTrack<Ref_Phase2TrackerDig
   std::array<float, 8> bendchi2_bins{{0, 0.5, 1.25, 2, 3, 5, 10, 50}};
   int n_bendchi2 = static_cast<int>(bendchi2_bins.size());
   float tmp_trk_bendchi2_bin = -1;
-  for (int i = 0; i < n_bendchi2; i++) {
+  for (int i = 0; i < (n_bendchi2 - 1); i++) {
     if (tmp_trk_bendchi2 >= bendchi2_bins[i] && tmp_trk_bendchi2 < bendchi2_bins[i + 1]) {
       tmp_trk_bendchi2_bin = i;
       break;
     }
   }
+  if (tmp_trk_bendchi2 >= bendchi2_bins[n_bendchi2 - 1])
```

Comment:
This moves all the overflows (tmp_trk_bendchi2 > 50.) into the 7th bin: was it intended?

Reply:
@perrotta @cecilecaillol The fix on L121 is good (loop out of array size) but L127 and L128 should be removed. We want all of the overflow to go into the 8th bin which is 50 - inf. L129 and L130 do this already by assigning it bin 8 if the bin has not been selected yet.